### PR TITLE
[LOOP-2385] Observe changes to glucose unit

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-2385

### CODE REVIEW

File: G4ShareSpy/G4CGMManagerSettingsNavigationController.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+//  Created by Nathaniel Hamming on 2021-01-15.
+//  Copyright © 2021 Mark Wilson. All rights reserved.
+//
+
+import UIKit
+import HealthKit
+import LoopKit
+import LoopKitUI
+
+class G4CGMManagerSettingsNavigationController: SettingsNavigationViewController, PreferredGlucoseUnitObserver {
```
Comment (darinkrauss) [inline G4ShareSpy/G4CGMManagerSettingsNavigationController.swift]: Recommend ditching this and using generic `CGMManagerSettingsNavigationController` described in my `LoopKit` PR comments.

**Review by darinkrauss — APPROVED** [review]
Approved, if you choose to pass on my recommendation. 🔭

File: G4ShareSpy/G4CGMManagerSettingsViewController.swift [hunk 2]
```diff
@@ -12,12 +12,11 @@ import LoopKitUI
 import ShareClient
 import ShareClientUI
 
-
 class G4CGMManagerSettingsViewController: UITableViewController {
 
     public let cgmManager: G4CGMManager
 
-    public let glucoseUnit: HKUnit
+    private var glucoseUnit: HKUnit
```
Comment (rickpasetto) [inline G4ShareSpy/G4CGMManagerSettingsViewController.swift:19]: [nit] prefer `let` for properties that don't mutate.

**Review by rickpasetto — APPROVED** [review]
LGTM.

**Pull request merged**